azure/services/natgateways/natgateways.go

```diff
@@ -35,22 +35,13 @@ type NatGatewayScope interface {
 	NatGatewaySpecs() []azure.ASOResourceSpecGetter[*asonetworkv1.NatGateway]
 }
 
-// Service provides operations on azure resources.
-type Service struct {
-	Scope NatGatewayScope
-	*aso.Service[*asonetworkv1.NatGateway, NatGatewayScope]
-}
-
 // New creates a new service.
-func New(scope NatGatewayScope) *Service {
+func New(scope NatGatewayScope) *aso.Service[*asonetworkv1.NatGateway, NatGatewayScope] {
```

Comment:
Could you please do the same for groups in this PR?

Reply:
I will raise a separate PR for it? What say?

Reply:
I don't see a reason that they need to be separate changes necessarily?

Reply:
Was suggesting a different PR because they are different services. Easier to revert just in case(this is a trivial change, wouldn't need to be reverted tho). 

But if you insist, happy to squash in "resource groups".

Reply:
I think it's probably more likely we'd have to revert both instead of just one, in which case both in one PR would be a little more handy.